server/routes/serviceProviderReferrals/dashboardPresenter.ts

```diff
@@ -18,12 +18,12 @@ export default class DashboardPresenter {
   ]
 
   readonly tableRows: SortableTableRow[] = this.referrals.map(referral => {
-    const { serviceCategoryId } = referral.referral
-    const serviceCategory = this.serviceCategories.find(aServiceCategory => aServiceCategory.id === serviceCategoryId)
-    if (serviceCategory === undefined) {
-      throw new Error(`Expected serviceCategories to contain service category with ID ${serviceCategoryId}`)
+    const interventionForReferral = this.interventions.find(
+      intervention => intervention.id === referral.referral.interventionId
+    )
+    if (interventionForReferral === undefined) {
+      throw new Error(`Expected referral to be linked to an intervention with ID ${referral.referral.interventionId}`)
```

Comment:
just an observation: it's a shame we can't add structured arguments to errors like we can with log messages

Reply:
also i think this error message could be improved. with no context i don't think it means a lot. how about something more descriptive like "couldn't populate dashboard row because the referral's intervention is missing from the list passed to the presenter" or something

Reply:
Have updated this error message, you're right - it wasn't that helpful. It's still impossible for it to occur, as we fetch those interventions from the referral's `interventionId`, but 🤷